chain/chunks/src/client.rs

```diff
@@ -36,20 +36,28 @@ pub struct ShardedTransactionPool {
     /// Useful to make tests deterministic and reproducible,
     /// while keeping the security of randomization of transactions in pool
     rng_seed: RngSeed,
+
+    /// If set, new transactions that bring the size of the pool over this limit will be rejected.
+    /// The size is tracked and enforced separately for each shard.
+    pool_size_limit: Option<u64>,
```

Comment:
Do you know what is the largest valid transaction?

Reply:
I've seen the number "4MiB" mentioned a few times, e.g. in https://github.com/near/nearcore/issues/8936 and https://github.com/near/nearcore/blob/e334efcca8a9228a28cc50a3143e4bb6dd5887c1/core/primitives/res/runtime_configs/parameters.snap#L157